### PR TITLE
[HOPS-1556] Added Test for handling RuntimeException in TransactionRe…

### PR DESCRIPTION
**Pull request by smkniazi**
…questHandler class

## Make sure there is no duplicate PR for this issue

* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added and passed (for bug fixes / features)
- [x] HOPS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit
- [x] The commit message has the following format: [HOPS-XXX] message

* **Post a link to the associated JIRA issue**
https://hopshadoop.atlassian.net/browse/HOPS-1556

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:
https://github.com/hopshadoop/hops-metadata-dal/pull/184

### CODE REVIEW

**Comment by berthoug** [comment]
Jenkins test this please.

File: hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestRunTimeExceptionInTxReqHandler.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+              }
+
+              @Override
+              public void acquireLock(TransactionLocks locks) throws IOException {
+              }
+            };
+    try {
+      txHandler.handle();
+    } catch (RuntimeException e) {
+      assert e.getMessage().compareToIgnoreCase("Test") == 0;
```
Comment (berthoug) [inline hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestRunTimeExceptionInTxReqHandler.java:75]: Are we expecting to get an exception? If yes should we fail if no excpetion is thrown?

File: hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestRunTimeExceptionInTxReqHandler.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      }
+
+      @Override
+      public void acquireLock(TransactionLocks locks) throws IOException {
+      }
+    };
+    try {
+      txHandler.handle();
+    } catch (RuntimeException e) {
+      assert e.getMessage().compareToIgnoreCase("Test") == 0;
```
Comment (berthoug) [inline hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestRunTimeExceptionInTxReqHandler.java:96]: Are we expecting to get an exception? If yes should we fail if no excpetion is thrown?

File: hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestRunTimeExceptionInTxReqHandler.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+      @Override
+      public void acquireLock(TransactionLocks locks) throws IOException {
+        throw new RuntimeException("Test");
+      }
+    };
+    try {
+      txHandler.handle();
+    } catch (RuntimeException e) {
+      assert e.getMessage().compareToIgnoreCase("Test") == 0;
```
Comment (berthoug) [inline hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestRunTimeExceptionInTxReqHandler.java:118]: Are we expecting to get an exception? If yes should we fail if no excpetion is thrown?

**Comment by berthoug** [comment]
The test got a null pointer in your new test: https://bbc1.sics.se/jenkins/job/hops-pull/441/io.hops$hadoop-hdfs/testReport/org.apache.hadoop.hdfs.server.namenode/TestRunTimeExceptionInTxReqHandler/TestSuppressTransientExceptions/

File: hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestRunTimeExceptionInTxReqHandler.java (hunk 1 above)
Comment (smkniazi) [reply hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestRunTimeExceptionInTxReqHandler.java:76]: Yes, we should. Fxd

File: hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestRunTimeExceptionInTxReqHandler.java (hunk 2 above)
Comment (smkniazi) [reply hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestRunTimeExceptionInTxReqHandler.java:98]: Yes, we should. Fxd

File: hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestRunTimeExceptionInTxReqHandler.java (hunk 3 above)
Comment (smkniazi) [reply hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/TestRunTimeExceptionInTxReqHandler.java:121]: Yes, we should. Fxd

**Comment by smkniazi** [comment]
> The test got a null pointer in your new test: https://bbc1.sics.se/jenkins/job/hops-pull/441/io.hops$hadoop-hdfs/testReport/org.apache.hadoop.hdfs.server.namenode/TestRunTimeExceptionInTxReqHandler/TestSuppressTransientExceptions/

Both TestSuppressTransientExceptions and TestRunTimeExceptionInTxReqHandler pass. Something is wrong with the Jenkins logs. Function name is garbage 
org.apache.hadoop.hdfs.server.namenode.TestRunTimeExceptionInTxReqHandler.__CLR4_4_0867wu2o4n(TestRunTimeExceptionInTxReqHandler.java:75)

**Comment by berthoug** [comment]
Ok, I'll run the tests again.

**Comment by berthoug** [comment]
Jenkins test this please.

**Comment by berthoug** [comment]
tests: https://bbc1.sics.se/jenkins/view/pull-manual/job/hops-pull-manual/123/testReport/

**Comment by smkniazi** [comment]
> tests: https://bbc1.sics.se/jenkins/view/pull-manual/job/hops-pull-manual/123/testReport/

 org.apache.hadoop.hdfs.TestReadWhileWriting.pipeline_02_03 [passes]	
 org.apache.hadoop.hdfs.server.balancer.TestBalancer.testTwoReplicaShouldNotInSameDN [passes]
 org.apache.hadoop.hdfs.server.balancer.TestBalancer.testBalancerWithPinnedBlocks [passes]
 org.apache.hadoop.hdfs.server.balancer.TestBalancer.testBalancerCliWithExcludeListWithPorts [passes]
 org.apache.hadoop.hdfs.server.namenode.TestFsck.testBlockIdCKDecommission [ passes ] 	
 org.apache.hadoop.hdfs.server.namenode.TestHABlockReports.testAllDeadNamenodes [passes]	


The fix was to rebase with master

**Pull request merged**